cmd/kubemark/hollow-node.go

```diff
@@ -227,3 +305,67 @@ func run(config *hollowNodeConfig) {
 		hollowProxy.Run()
 	}
 }
+
+func newEKSAuthProvider(_ string, config map[string]string, _ restclient.AuthProviderConfigPersister) (restclient.AuthProvider, error) {
```

Comment:
All the code below this line seems to be eks-specific. I think we shouldn't put provider-specific things inside this generic hollow-node code. Can you think of some way in which we can abstract away these functions into cloud-provider code? (maybe put it in https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/aws)